diff --git a/vendor/ingenico/library-ops-m2/Controller/Retry/Payment.php b/Controller/Retry/Payment.php
index 83a8511..abe103c 100644
--- a/vendor/ingenico/library-ops-m2/Controller/Retry/Payment.php
+++ b/vendor/ingenico/library-ops-m2/Controller/Retry/Payment.php
@@ -24,6 +24,8 @@ use Magento\Sales\Api\Data\OrderPaymentInterface;
 use Magento\Sales\Model\Order;
 use Magento\Sales\Model\OrderRepository;
 use Netresearch\OPS\Model\Retry\GeneratorInterface;
+use Magento\Checkout\Model\Session as CheckoutSession;
+
 
 /**
  * Class Payment
@@ -42,15 +44,21 @@ class Payment extends RetryAbstract
      */
     private $customerSession;
 
+    /**
+    * @var CheckoutSession
+    */
+    private $checkoutSession;
+
     /**
      * Retry constructor.
      *
      * @param Context $context
-     * @param Registry $registry
      * @param OrderRepository $orderRepository
      * @param SearchCriteriaBuilder $criteriaBuilder
-     * @param Session $customerSession
      * @param GeneratorInterface $retryPage
+     * @param Registry $registry
+     * @param Session $customerSession
+     * @param CheckoutSession $checkoutSession
      */
     public function __construct(
         Context $context,
@@ -58,7 +66,8 @@ class Payment extends RetryAbstract
         SearchCriteriaBuilder $criteriaBuilder,
         GeneratorInterface $retryPage,
         Registry $registry,
-        Session $customerSession
+        Session $customerSession,
+        CheckoutSession $checkoutSession
     ) {
         parent::__construct(
             $context,
@@ -68,6 +77,7 @@ class Payment extends RetryAbstract
         );
         $this->registry = $registry;
         $this->customerSession = $customerSession;
+        $this->checkoutSession = $checkoutSession;
     }
 
     /**
@@ -83,6 +93,7 @@ class Payment extends RetryAbstract
             $message = __('Hash not valid');
         } elseif ($this->canRetryPayment($payment) && $order->getState() !== Order::STATE_CANCELED) {
             $this->registry->register('current_order', $order, true);
+            $this->checkoutSession->setQuoteId($order->getQuoteId());
 
             if (!$order->getCustomerIsGuest()) {
                 $this->customerSession->setCustomerId($order->getCustomerId());
